snakemake/persistence.py

```diff
@@ -61,6 +63,18 @@ def __init__(
         # place to store any auxiliary information needed during a run (e.g. source tarballs)
         self.aux_path = os.path.join(self.path, "auxiliary")
 
+        self.incomplete_oldapproach = os.path.exists(
+            self._metadata_path
+        ) and not os.path.exists(self._incomplete_path)
```

Comment:
I think if the incomplete folder exists, we can always use the new approach. If it doesn't exist, we can do a one-time migration, by scanning through the metadata path and generating a flag file for each incomplete item in the incomplete path. Then, we can always use the new self.incomplete() version.